src/diffcalc_API/routes/constraints.py

```diff
@@ -1,52 +1,73 @@
-from typing import Dict, Union
+from typing import Dict, Optional, Union
 
-from fastapi import APIRouter, Body, Depends, Response
+from fastapi import APIRouter, Body, Depends, Query, Response
 
 from diffcalc_API.services import constraints as service
-from diffcalc_API.stores.pickling import get_store
+from diffcalc_API.stores.mongo import get_store
 from diffcalc_API.stores.protocol import HklCalcStore
 
 router = APIRouter(prefix="/constraints", tags=["constraints"])
 
 
 @router.get("/{name}")
-async def get_constraints(name: str, store: HklCalcStore = Depends(get_store)):
-    content = await service.get_constraints(name, store)
-
+async def get_constraints(
+    name: str,
+    store: HklCalcStore = Depends(get_store),
+    collection: Optional[str] = Query(default=None, example="B07"),
+):
+    content = await service.get_constraints(name, store, collection)
     return Response(content=content, media_type="application/text")
```

Comment:
I reckon they're better suited in the URL, but it's possible I don't have the correct mental model of how the service will be used

Reply:
Ideally, yes these would go into a seperate PR. However I've already made changes in this branch, I feel like just going ahead with implementing it how I've done it, and then making another PR if I want to change it to this method.

The reason I thought a query was better for collection, is so that there is a default collection for users who dont want their crystals associated with a specific beamline. However, this sounds messy and I don't know how practical a use case it is.

Reply:
I think I'll make a separate PR with this. I'll just keep the collection as a query for now.